src/CMakeLists.txt

```diff
@@ -5,8 +5,14 @@ project(minetest)
 INCLUDE(CheckIncludeFiles)
 INCLUDE(CheckLibraryExists)
 
+if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
```

Comment:
Should be moved into the `else # GCC or compatible compilers such as Clang` about ~600 lines down.